commons/src/gemfile_lock.rs

```diff
@@ -116,12 +116,12 @@ impl FromStr for GemfileLock {
     type Err = std::convert::Infallible;
 
     fn from_str(string: &str) -> Result<Self, Self::Err> {
```

Comment:
In the Slack thread, I see the error message when someone tried to use the pre-release version was an HTTP 404 rather than an error message that mentioned finding an invalid version in `Gemfile.lock`. 

Longer term, would it be worth having more explicit handling for invalid versions, so that they are rejected early with a specific "invalid version; here's the form it should take" type error, rather than the buildpack only realising its invalid via the 404?

For example, the Python CNB shows this specific error message (and in this case never tries to make an HTTP request):
https://github.com/heroku/buildpacks-python/blob/e62672366eb192750b166c2af883e10b43cbac54/tests/python_version_test.rs#L207-L222

Reply:
Currently the Ruby buildpack doesn't know what is and isn't a valid version. It takes whatever you give it and sees if it's on S3. Longer term the idea is to move to an inventory file model, but the going is slow. I'm currently able to generate inventory updates, but I need a way to bundle them all together and make a self PR. I poked at it briefly https://github.com/heroku/docker-heroku-ruby-builder/commit/204df94b3263e0340a57f08295e6b6c49d77a4e0.

Once we've got an inventory file we could levenshtein distance and say "X.Q.P" is not valid "did you mean X.Y.Z?"

The classic buildpack suggests versions by querying/walking S3, which... if we're moving to an inventory file, isn't worth implementing here if we're just going to turn around and throw it away in favor of a file.